climada/hazard/tc_tracks_synth.py

```diff
@@ -146,22 +207,28 @@ def calc_perturbed_trajectories(tracks,
     tracks.data = sum(new_ens, [])
 
     if decay:
-        hist_tracks = [track for track in tracks.data if track.orig_event_flag]
-        if hist_tracks:
-            try:
-                extent = tracks.get_extent()
-                land_geom = climada.util.coordinates.get_land_geometry(
-                    extent=extent, resolution=10
-                )
-                v_rel, p_rel = _calc_land_decay(hist_tracks, land_geom,
-                                                pool=tracks.pool)
-                tracks.data = _apply_land_decay(tracks.data, v_rel, p_rel,
-                                                land_geom, pool=tracks.pool)
-            except ValueError:
-                LOGGER.info('No land decay coefficients could be applied.')
+        extent = tracks.get_extent()
+        land_geom = climada.util.coordinates.get_land_geometry(
+            extent=extent, resolution=10
+        )
+        if use_global_decay_params:
+            tracks.data = _apply_land_decay(tracks.data, LANDFALL_DECAY_V,
+                                            LANDFALL_DECAY_P,
+                                            land_geom, pool=tracks.pool)
         else:
-            LOGGER.error('No historical tracks contained. '
-                         'Historical tracks are needed for land decay.')
+            # fit land decay coefficients based on historical tracks
+            hist_tracks = [track for track in tracks.data if track.orig_event_flag]
+            if hist_tracks:
+                try:
+                    v_rel, p_rel = _calc_land_decay(hist_tracks, land_geom,
+                                                    pool=tracks.pool)
+                    tracks.data = _apply_land_decay(tracks.data, v_rel, p_rel,
+                                                    land_geom, pool=tracks.pool)
+                except ValueError:
+                    LOGGER.info('No land decay coefficients could be applied.')
```

Comment:
If you catch an error, you probably should also handle it... or at least the logging message must be very clear. Why could no land decay coefficients be applied? What does it mean?

Reply:
There could be several reasons why it failed. In a new commit I now raise an error, re-raising the error that was caught, too (with `from`) to help the user identifying the cause.